modules/highgui/cmake/detect_gtk.cmake

```diff
@@ -1,9 +1,13 @@
 # --- GTK ---
-ocv_clear_vars(HAVE_GTK HAVE_GTK2 HAVE_GTK3 HAVE_GTKGLEXT)
+ocv_clear_vars(HAVE_GTK HAVE_GTK2 HAVE_GTK3 HAVE_GTK4 HAVE_GTKGLEXT)
 if(WITH_GTK)
   if(NOT WITH_GTK_2_X)
+    ocv_check_modules(GTK4 gtk4)
     ocv_check_modules(GTK3 gtk+-3.0)
```

Comment:
With QT we have a way to specify major version: `-DWITH_QT=4` or `-DWITH_QT=5`.
Perhaps we should go in this way here too.